src/main/java/io/cryostat/jfr/datasource/server/JfrResource.java

```diff
@@ -223,6 +244,25 @@ private void logUploadedFile(String file, StringBuilder responseBuilder) {
         LOGGER.info("Uploaded: " + file);
     }
 
+    private synchronized void setLoadedFile(String filename) {
+        CompletableFuture<Void> taskFuture =
+                CompletableFuture.runAsync(
+                        () -> {
+                            try {
+                                loadedFile = filename;
```

Comment:
I'm not sure I follow why this piece of code needs to be run async. It's just setting a value to an instance field, right? That's a non-blocking operation, so it can just be done synchronously.

Reply:
Yeah, now that I think about it, it doesn't really make sense. I was thinking about how the HTTP requests could be sent at the same time and the /list request may give a wrong value, but that doesn't make sense to run async.